### PR TITLE
Add build info and version info to TSDB at build time

### PR DESCRIPTION
**Pull request by imakhlin**
Example:
$ **tsdbctl ver**
```
tsdbctl build information:
  Build time: 2018-11-12T11:48:34+0200
  OS: darwin
  Architecture: amd64
  Version: f970431
  SHA: f970431b3aef562119be3d360f1a8c06e963c60a
  Branch: tsdb.revision
```

### CODE REVIEW

**Comment by gshatz** [comment]
Like!
is there a way for us to invoke this from nuclio?

File: pkg/config/config.go [hunk 1]
```diff
@@ -56,10 +56,31 @@ const (
 	DefaultVerboseLevel           = "debug"
 )
 
+type BuildInfo struct {
+	BuildTime    string `json:"os,omitempty"`
+	Os           string `json:"os,omitempty"`
```
Comment (gtopper) [inline pkg/config/config.go]: C&P error in json tag.

File: Makefile [hunk 2]
```diff
@@ -14,6 +20,18 @@ GOPATH ?= $(shell go env GOPATH)
...
 
+BUILD_TIME := $(shell date +%FT%T%z)
+CONFIG_PKG=github.com/v3io/v3io-tsdb/pkg/config
+
+BUILD_OPTS := -ldflags " \
+  -X $(CONFIG_PKG).osys=$(GOOS) \
+  -X $(CONFIG_PKG).architecture=$(GOARCH) \
+  -X $(CONFIG_PKG).version=$(GIT_REVISION) \
+  -X $(CONFIG_PKG).revision=$(GIT_SHA_SHORT) \
+  -X $(CONFIG_PKG).branch=$(GIT_BRANCH) \
```
Comment (gtopper) [inline Makefile]: Best wrap every value with quotes. Better safe than soggy ☔️

File: pkg/config/config.go [hunk 3]
```diff
@@ -56,10 +56,31 @@ const (
...
 	once     sync.Once
 	failure  error
+
+	BuildMetadta = &BuildInfo{
+		BuildTime:    buildTime,
+		Os:           osys,
+		Architecture: architecture,
+		Version:      version,
+		Revision:     revision,
+		Branch:       branch,
```
Comment (gtopper) [inline pkg/config/config.go]: I know it's a minor thing, but can we have these in the same order everywhere? 😅

File: pkg/config/config.go [hunk 4]
```diff
@@ -111,6 +132,8 @@ type V3ioConfig struct {
 	// Don't aggregate from raw chunks, for use when working as a Prometheus
 	// TSDB library
 	DisableClientAggr bool `json:"disableClientAggr,omitempty"`
+	// Build Info
+	BuildInfo *BuildInfo `json:"buildInfo,omitempty"`
```
Comment (gtopper) [inline pkg/config/config.go]: I think not only can we ditch the pointer, this could be a convenient place to use struct "inheritance" notation.

Comment (gtopper) [reply pkg/config/config.go]: e.g.:
```
type A struct {
  B
  extra uint64
}
```

Comment (gtopper) [reply pkg/config/config.go]: But not all that important.

File: pkg/tsdbctl/version.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	rootCommandeer *RootCommandeer
+}
+
+func newVersionCommandeer(rc *RootCommandeer) *versionCommandeer {
+	commandeer := &versionCommandeer{
+		rootCommandeer: rc,
+	}
+
+	cmd := &cobra.Command{
+		Aliases: []string{"ver"},
```
Comment (gtopper) [inline pkg/tsdbctl/version.go]: Very lazy alias 😛

File: pkg/tsdbctl/version.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		Use:     "version",
+		Hidden:  false,
+		Short:   "Displays version",
+		Long:    "Displays version information",
```
Comment (gtopper) [inline pkg/tsdbctl/version.go]: I think we don't need a short and a long here if the difference between them is going to be the word "information".

**Comment by imakhlin** [comment]
@Sharon-iguazio Please review the wording

**Pull request merged**